src/plugin/non_instantiable_plugin.cpp

```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/abstract_plugin.hpp"
+#include "utils/singleton.hpp"
+
+namespace opossum {
+
+// This plugin does not export its instantiation so that we can test if this case is handled correctly.
+class NonInstantiablePlugin : public AbstractPlugin, public Singleton<NonInstantiablePlugin> {
```

Comment:
Put "test" in the class name so that it is clear that it is not supposed to be used on the outside.